src/components/ProductSearch/Products.tsx

```diff
@@ -1,24 +1,30 @@
 import { Dispatch, SetStateAction, useEffect } from "react";
 import ProductGrid from ".";
-import { ProductsType } from "@/service/types";
+import { ProductType, ProductsType } from "@/service/types";
+import { useFetch } from "@/hooks/useFetch";
 
 type ProductsProps = {
     products: ProductsType;
     setProducts: Dispatch<SetStateAction<ProductsType>>;
+    url: string;
 };
 
-export const Products = ({ products, setProducts }: ProductsProps) => {
+export const Products = ({ products, setProducts, url }: ProductsProps) => {
+    const { data, error } = useFetch<ProductsType>(url);
```

Comment:
Here you could import `base_url` and `products`

```suggestion
    const { data, error } = useFetch<ProductsType>(`${base_url}${products}`);
```